### PR TITLE
feature call

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 53553: Create UI](https://edu-redmine.sun-asterisk.vn/issues/53553)
- [#Task 53554: Handle event](https://edu-redmine.sun-asterisk.vn/issues/53554)
- [#Task 53555: get data](https://edu-redmine.sun-asterisk.vn/issues/53555)
- [#Task 53557: Create UI](https://edu-redmine.sun-asterisk.vn/issues/53557)
- [#Task 53558: Handle event](https://edu-redmine.sun-asterisk.vn/issues/53558)
- [#Task 53559: get data](https://edu-redmine.sun-asterisk.vn/issues/53558)
## WHAT
- send invite call
- imcoming screen
- outgoing screen
- call screen
## Evidence (Screenshot or Video)
image
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

**Review by ndt0409 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/example/finalapplication/Utils/Constant.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ package com.example.finalapplication.utils
 import java.util.*
 
 object Constant {
+    const val ERROR_USER_NOT_CONNECT = "Không thể liên lạc được"
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/Utils/Constant.kt]: đổi sang tiếng anh đi e

File: app/src/main/java/com/example/finalapplication/Utils/ApiFcmService.kt [hunk 2]
```diff
@@ -11,6 +11,6 @@ interface ApiFcmService {
     @POST("send")
     fun sendRemoteMessage(
         @HeaderMap headers: HashMap<String, String>,
-        @Body body: Message
-    ): Call<Message>
+        @Body body: Notification
+    ): Call<Notification>
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/Utils/ApiFcmService.kt]: dùng coroutine , bỏ call đi nhé

File: app/src/main/java/com/example/finalapplication/data/model/Notification.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import android.os.Parcelable
+import com.google.gson.annotations.SerializedName
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+data class Notification(
+    @SerializedName("data")
+    val data: DataNotification,
+    @SerializedName("registration_ids")
+    val registrationIds: List<String?>) : Parcelable
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/data/model/Notification.kt]: tất cả model dùng Gson parse phải check null hết nhé

File: app/src/main/java/com/example/finalapplication/screen/chatroom/ChatRoomActivity.kt [hunk 4]
```diff
@@ -43,12 +48,21 @@ class ChatRoomActivity : BaseActivity<ActivityChatRoomBinding>(ActivityChatRoomB
...
-                    currentUser.id.toString(),
+                    currentUser?.id.toString(),
                     adversaryUser?.id.toString()
                 )
                 message.image = uri.toString()
-                message.roomId = currentUser.id + adversaryUser?.id
-                adversaryUser?.let { it1 -> chatViewModel.sendMessage(currentUser, it1, message) }
+                message.roomId = currentUser?.id + adversaryUser?.id
+                isLoadImage = true
+                adversaryUser?.let { it1 ->
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/finalapplication/screen/chatroom/ChatRoomActivity.kt]:
```suggestion
                adversaryUser?.let { user ->
```
ko chơi it1 ,it2 đâu nhé

**Pull request merged**